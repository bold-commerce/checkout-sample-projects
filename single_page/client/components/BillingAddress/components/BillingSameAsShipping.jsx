import { RadioField } from '@boldcommerce/stacks-ui';
import React from 'react';
import { useTranslation } from 'react-i18next';

const BillingSameAsShipping = ({
  billingSameAsShipping,
  setBillingSameAsShipping,
  disabled,
}) => {
  const { t } = useTranslation();
  return (
  <>
    <RadioField
      label={t('billing.same_as_shipping')}
      name="billing-address"
      className="RadioButton"
      value="SAME_AS_SHIPPING_ADDRESS"
      checked={billingSameAsShipping}
      onChange={() => setBillingSameAsShipping(true)}
      disabled={disabled}
    />
    <RadioField
      label={t('billing.different_from_shipping')}
      name="billing-address"
      className="RadioButton"
      value="DIFFERENT_BILLING_ADDRESS"
      checked={!billingSameAsShipping}
      onChange={() => setBillingSameAsShipping(false)}
      disabled={disabled}
    />
  </>
  );
};

export default BillingSameAsShipping;
