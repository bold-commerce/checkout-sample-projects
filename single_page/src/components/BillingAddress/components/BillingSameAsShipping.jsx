import { RadioField } from '@boldcommerce/stacks-ui';
import React from 'react';

const BillingSameAsShipping = ({
  billingSameAsShipping,
  setBillingSameAsShipping,
  disabled,
}) => (
  <>
    <RadioField
      label="Same as shipping address"
      name="billing-address"
      className="RadioButton"
      value="SAME_AS_SHIPPING_ADDRESS"
      checked={billingSameAsShipping}
      onChange={() => setBillingSameAsShipping(true)}
      disabled={disabled}
    />
    <RadioField
      label="Use a different billing address"
      name="billing-address"
      className="RadioButton"
      value="DIFFERENT_BILLING_ADDRESS"
      checked={!billingSameAsShipping}
      onChange={() => setBillingSameAsShipping(false)}
      disabled={disabled}
    />
  </>
);

export default BillingSameAsShipping;
