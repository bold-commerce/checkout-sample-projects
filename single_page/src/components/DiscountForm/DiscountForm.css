.DiscountForm {
  display: flex;
  padding: 20px 0;
  border-top: 1px solid rgba(0,0,0,0.1);
  border-bottom: 1px solid rgba(0,0,0,0.1);
}

.DiscountForm .stx-field {
  flex: 2;
  margin: 0;
}

.DiscountForm button.stx-button .stx-button__text {
  text-transform: capitalize;
}

.DiscountForm button.stx-button {
  flex: 0.5;
  height: 42px;
}

.Summary__DiscountForm .DiscountForm__error {
  color: #FF1A2D;
}

@media screen and (max-width: 599px) {
  .DiscountForm {
    flex-direction: column;
  }

  .DiscountForm .stx-field {
    flex: 1;
    margin: 0 0.25rem 10px 0.25rem;

  }

  .DiscountForm button.stx-button {
    flex: 1
  }
}
