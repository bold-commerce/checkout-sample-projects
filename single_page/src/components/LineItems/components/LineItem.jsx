import React from 'react';
import PropTypes from 'prop-types';
import Product from './Product';
import { Price } from '@boldcommerce/stacks-ui';
import { useTranslation } from 'react-i18next';

const LineItem = ({
  title,
  image,
  quantity,
  totalPrice,
  variants
}) => {
  const { t } = useTranslation();
  return(
    <div className="SummaryBlock CartItem" >
      <Product
        title={title}
        image={image}
        variants={variants}
      />
      <div className="CartItem__QuantityPriceWrapper">
        <div className="CartItem__ProductQuantityWrapper">
          <div className="ProductQuantity" aria-label="product quantity">
            {quantity}
          </div>
        </div>
        <div className="CartItem__ProductPrice">
          <Price amount={totalPrice} moneyFormatString={t('currency_format')} />
        </div>
      </div>
    </div>
  );}
  
  LineItem.propTypes = {
    title: PropTypes.string,
    image: PropTypes.string,
    quantity: PropTypes.number,
    lineItemKey: PropTypes.string,
    variants: PropTypes.array,
  };
  
  export default LineItem;