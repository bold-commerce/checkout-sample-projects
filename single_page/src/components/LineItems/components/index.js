export { default as LineItem } from './LineItem';
export { default as Product } from './Product';
