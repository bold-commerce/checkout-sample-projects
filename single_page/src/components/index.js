export * from './Address';
export * from './BillingAddress';
export * from './CheckoutButton';
export * from './CheckoutSection';
export * from './Customer';
export * from './DiscountForm';
export * from './EmptyState';
export * from './Header';
export * from './LineItems';
export * from './LoadingState';
export * from './OrderErrors';
export * from './OrderSummary';
export * from './PaymentMethod';
export * from './ShippingAddress';
export * from './ShippingLines';
