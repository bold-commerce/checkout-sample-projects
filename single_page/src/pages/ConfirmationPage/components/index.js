export * from './ConfirmationList';
export * from './ConfirmationListItem';
