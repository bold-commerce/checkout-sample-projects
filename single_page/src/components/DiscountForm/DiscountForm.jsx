import { useDiscount } from '@boldcommerce/checkout-react-components';
import { Button, InputField } from '@boldcommerce/stacks-ui';
import React, { memo, useCallback, useState } from 'react';
import { useAnalytics, useErrorLogging } from '../../hooks';
import './DiscountForm.css';

const DiscountForm = () => {
  const { data, applyDiscount } = useDiscount();
  const {
    discountApplied,
    discountCode,
  } = data;

  return (
    <MemoizedDiscountForm 
      discountCode={discountCode}
      discountApplied={discountApplied}
      applyDiscount={applyDiscount}
    />
  );
};

const MemoizedDiscountForm = memo(({
  discountCode,
  discountApplied,
  applyDiscount,
}) => {
  const trackEvent = useAnalytics();
  const logError = useErrorLogging();
  const [discount, setDiscount] = useState(discountCode);
  const [errors, setErrors] = useState(null);
  const [loading, setLoading] = useState(false);

  const handleApply = useCallback(async () => {
    setLoading(true);
    try {
      await applyDiscount(discount);
      trackEvent('apply_discount_code');
      setErrors(null);
    } catch(e) {
      logError('discount_code', e);
      if (e.body.errors[0].type !== 'authorization.expired_jwt') {
        setErrors(e.body.errors);
      }
    }
    setLoading(false);
  }, [discount, applyDiscount, logError, trackEvent]);

  return (
    <div className="SummaryBlock Summary__DiscountForm">
      <div className="DiscountForm">
        <InputField
          type="text"
          placeholder="Enter discount code"
          value={discount}
          messageText={errors && errors[0].message}
          messageType={errors && 'alert'}
          onChange={(e) => setDiscount(e.target.value)}
          disabled={discountApplied || loading}
        />
        <Button
          primary={discountApplied || discount.length > 0}
          disabled={discount.length === 0 || discountApplied || loading}
          onClick={handleApply}
          loading={loading}
        >
          Apply
        </Button>
      </div>
    </div>
  );
});

export default DiscountForm;
