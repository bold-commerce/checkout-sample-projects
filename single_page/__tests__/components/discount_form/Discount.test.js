import React from "react";
import { render } from "@testing-library/react";
import { DiscountForm } from '../../../client/components/DiscountForm';

jest.mock('@boldcommerce/checkout-react-components', () => ({
  useDiscount: () => ({
    data: {
      discountApplied: true,
      discountCode: 'Test'
    },
    applyDiscount: (() => {})
  })
}))


describe('DiscountForm', () => {
  test('renders DiscountForm component', () => {
    const { asFragment } = render( <DiscountForm/>)

    expect(asFragment()).toMatchSnapshot();
  })
})
