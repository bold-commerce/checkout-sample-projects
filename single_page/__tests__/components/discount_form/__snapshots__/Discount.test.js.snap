// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscountForm renders DiscountForm component 1`] = `
<DocumentFragment>
  <div
    class="SummaryBlock Summary__DiscountForm"
  >
    <div
      class="DiscountForm"
    >
      <div
        class="stx-field stx-field--is-disabled stx-field--with-input"
      >
        <div
          class="stx-input stx-input--is-disabled"
        >
          <input
            aria-disabled="true"
            class="stx-input__input-element"
            disabled=""
            id="stxField1"
            placeholder="Enter discount code"
            type="text"
            value="Test"
          />
        </div>
      </div>
      <button
        aria-busy="false"
        aria-disabled="true"
        class="stx-button stx-button--primary stx-button--is-disabled"
        disabled=""
      >
        <span
          class="stx-button__text"
        >
          Apply
        </span>
        <span
          class="stx-loading-spinner stx-button__spinner"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
